perllib/FixMyStreet/Cobrand/Smidsy.pm

```diff
@@ -416,6 +416,307 @@ sub reports_hook_restrict_bodies_list {
     }
 }
 
+sub loop_period {
```

Comment:
As this is a copy (slightly extended for generic `$end`) of the method in UpdateAllReports, could we put it in core somewhere both can use it?